src/edu/umass/cs/gnscommon/utils/CanonicalJSON.java

```diff
@@ -118,7 +118,16 @@ private static String renderSimpleCanonicalJSON(Object x) {
           throw new JSONException("Bad value from toJSONString: " + object);
         }
         if (x instanceof Number) {
-          return JSONObject.numberToString((Number) x);
+          if (x instanceof Double || x instanceof Float) {
+            // This is a workaround for the issue described in MOB-1103 where the iOS and 
+            // potentially other clients have different printed representations for floating point numbers.
+            // This effectively removes floats from the canonical json string that is returned.
+            // A similar patch will be made on the ios client.
+            return "float";
```

Comment:
If I understand correctly, this change produces something like 
`{"data":{"lat":"float","lon":"float", "name":"westy"}}` from 
`{"data":{"lat": 12.72999965,"lon": 77.977742338, "name":"westy"}}`, right?